wandb/util.py

```diff
@@ -1426,3 +1430,31 @@ def _log_thread_stacks():
             logger.info('  File: "%s", line %d, in %s' % (filename, lineno, name))
             if line:
                 logger.info("  Line: %s" % line)
+
+
+def check_dict_contains_artifact(d):
+    if isinstance(d, dict):
+        for _, item in six.iteritems(d):
+            if isinstance(item, dict):
+                contains_artifacts = check_dict_contains_artifact(item)
+                if contains_artifacts:
+                    return True
+            elif isinstance(item, wandb.Artifact) or isinstance(
+                item, wandb.apis.public.Artifact
+            ):
+                return True
+    return False
+
+
+def convert_artifact_to_json_config(arti):
```

Comment:
It seems like this business logic should go in the Artifact class itself, and invoked before calling `json_friendly`.